module/app/app.go

```diff
@@ -906,18 +919,35 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	return paramsKeeper
 }
 
+// Registers handlers for all our upgrades
 func (app *Gravity) registerUpgradeHandlers() {
-	app.upgradeKeeper.SetUpgradeHandler("v2", func(ctx sdk.Context, plan upgradetypes.Plan, _ module.VersionMap) (module.VersionMap, error) {
-		// 1st-time running in-store migrations, using last version as fromVersion to
-		// avoid running InitGenesis.
-		fromVM := make(map[string]uint64)
-		for moduleName, module := range app.mm.Modules {
-			fromVM[moduleName] = module.ConsensusVersion()
-		}
-
-		// set gravity version to 1 in order to execute migration scripts from 1 to 2
-		fromVM[gravitytypes.StoreKey] = 1
+	upgrades.RegisterUpgradeHandlers(
+		app.mm, app.configurator, app.accountKeeper, app.bankKeeper, app.bech32IbcKeeper, app.distrKeeper,
```

Comment:
at first glance I notice this doesn't include the gravity upgrade handler, obviously this isn't needed given our testing, but I have no idea why.

Reply:
This section in app.go is for the chain upgrade, which involves upgrades across modules and chain specific issues. If we had an issue where some gravity module state needed to be modified and we write custom code for that, we'd need to add that. Examples could be initializing module state (like bech32ibc), fixing a mistake that happened on chain (like an airdrop), or data in one module affecting another (like the genesis changes I made for the last major upgrade).

The within-module migrations are all handled separately. Things like changing the format of data, adding defaults for new fields. They get registered by their module.go file and the app's module manager will call those at the end of the upgrade handler, when we call app.mm.RunMigrations()